src/ParallelAlgorithms/Actions/InitializeItems.hpp

```diff
@@ -0,0 +1,75 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <optional>
+#include <utility>
+
+#include "DataStructures/DataBox/DataBox.hpp"
+#include "Parallel/AlgorithmExecution.hpp"
+#include "Utilities/TMPL.hpp"
+#include "Utilities/TaggedTuple.hpp"
+
+/// \cond
+namespace Parallel {
+template <typename Metavariables>
+class GlobalCache;
+}  // namespace Parallel
+/// \endcond
+
+namespace Initialization {
+namespace Actions {
+/*!
+ * \ingroup ActionsGroup
+ * \brief Mutate DataBox items by calling db::mutate_apply on each Mutator in
+ *  the order they are specified
+ *
+ * There's a specialization for `InitializeItems<tmpl::list<Mutators...>>` that
+ * can also be used if a `tmpl::list` is available.
+ *
+ * \details In addition to the requirements specified by db::mutate_apply, each
+ * Mutator must define the type aliases of this action.
+ */
+template <typename... Mutators>
+struct InitializeItems {
```

Comment:
Do you intend to remove the existing `MutateApply` action? That's a special case of this one (single mutator). Currently CCE uses it mostly, which should probably move to just calling functions.

Reply:
Currently my plan is to convert initialization actions into mutators passed to this.  I expect  the actions `AddSimpleTags` and `AddComputeTags` will end up being removed by this process.  And any use of `MutateApply` in the list of initialization actions will be eliminated.

@nilsdeppe and I have long ago come to the conclusion that CCE needs to be refactored into fewer actions and more functions. This would lead to fewer things stored in the CCE DataBoxes, leading to improved compilation time and memory usage.    But currently this isn't high on my priority list.